src/squashdelta.cxx

```diff
@@ -157,8 +174,7 @@ std::list<struct compressed_block> get_blocks(MMAPFile& f, Compressor*& c,
 				if (block_list[j] & squashfs::block_size::uncompressed)
 				{
 					// seek over the uncompressed block
-					uint32_t len = (block_list[j]
-							& ~squashfs::block_size::uncompressed);
+					uint32_t len = (block_list[j] & ~squashfs::block_size::uncompressed);
```

Comment:
In this loop (3 lines above), if `block_count` is now `uint64_t`. should we declare the iterator `j` as `uint64_t` too?
Otherwise `for (uint32_t j = 0; j < block_count; ++j)` could overflow `j` and potentially cause an infinite loop.